README.md

```diff
@@ -0,0 +1,131 @@
+# DAI Wormhole
+
+DAI Wormhole facility allows users to fast teleport DAI between "domains", i.e. different chains that have a settlement mechanism with Ethereum L1. If DAI is teleported from L2 -> L1, this is equvialent to "fast withdrawal". DAI will be minted on L1 as soon as the transaction is confirmed on L2 and sent to the user, and when transaction eventually settles on L1, DAI will be released from L1 Bridge escrow (to have DAI on L2 in the first place, it had to be put on L1 escrow some time before) and burned. 
+
+If DAI is teleported from L2 -> L2, on the source domain it will be burned and on the destination domain it will be minted, while settlement process on L1 will eventually move DAI from source domain bridge escrow to destination domain bridge escrow.
+
+## Roles
+
+* **Initiator** - person initiating DAI transfer by calling `initiateWormhole` . They can optionally specify Operator and Receiver 
+* **Operator** - person (or specified third party) responsible for initiating minting process on destination domain by providing (in the fast path) Oracle attestations. Can call `requestMint` on `WormholeOracleAuth`
+* **Receiver** - person receiving minted DAI on a destination domain
+
+## DAI Wormhole L2 → L1 (aka fast withdrawals) 
+
+![FastWithdrawal](./docs/fw.png?raw=true)
+
+
+### Normal (fast) path
+
+To fast withdraw DAI from L2, user:
+
+* Calls `l2bridge.initiateWormhole()` - this burns DAI on L2 and sends `finalizeRegisterWormhole()` L2 -> L1 message to withdraw DAI from L2 bridge. This message, in normal cicumstances, will never be relayed and it will eventually expire in L1 message queue
```

Comment:
This (same for next sections) should be ordered list

Reply:
Ordered, nested lists don't seem to work, leaving as it is right now